src/components/ModelTableList/CloudGroup.js

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import MainTable from "@canonical/react-components/dist/components/MainTable";
+import {
+  generateStatusElement,
+  getModelStatusGroupData,
+  extractOwnerName
+} from "app/utils";
+import { getGroupedModelDataByCloud } from "app/selectors";
+import { generateModelDetailsLink, getStatusValue } from "./shared";
+
+/**
+  Returns the model info and statuses in the proper format for the table data.
+  @param {Object} groupedModels The models grouped by state
+  @returns {Object} The formatted table data.
+*/
+function generateModelTableDataByCloud(groupedModels) {
+  const modelData = {};
+  Object.keys(groupedModels).forEach(owner => {
```

Comment:
Shouldn't this be long over the clouds?

Reply:
I think the issue here is just that `owner` is actually `cloud`